apps/common/behaviours/generate-pdf.js

```diff
@@ -0,0 +1,50 @@
+'use strict';
+
+const pdf = require('html-pdf');
+const UploadModel = require('../../common/models/file-upload');
+const concat = require('concat-stream');
+const path = require('path');
+const config = require('../../../config');
+
+module.exports = superclass => class extends superclass {
+  process(req, res, next) {
+    this.generatePDF(req, res, (pdfBuffer) => {
+      const file = {
+        name: 'application_form.pdf',
+        data: pdfBuffer,
+        mimetype: 'application/pdf'
+      };
+      const model = new UploadModel(file);
+      model.save()
+        .then((result) => {
+          req.form.values['pdf-upload'] = result.url;
+        })
+        .then(() => {
+          super.process(req, res, next);
+        }, next);
+    });
+  }
+
+  generatePDF(req, res, next) {
+    res.render('pdf.html', Object.assign({}, this.locals(req, res), {
+      title: 'Firearms Application'
+    }), (err, html) => {
+      if (err) {
+        next(err);
+      }
+      let concatStream = concat(next);
+      // phantom is weird about paths it doesn't read things like a webserver
+      html = html.replace(/href="\//g, 'href="./');
+      pdf.create(html, {
+        base: 'file://' + path.resolve(__dirname, '../../../') + '/',
```

Comment:
Presuming this points to the folder containing, images, css and js. It should be local to the behaviour, not the <project>/assets and scss compiled dynamically if necessary.